Full `pytest` fails on no-device hosts and current Rich-rendered help

### What happened?

The documented local check (`pytest`) is not deterministic on a clean host without an attached iPhone/usbmuxd:

- Device fixtures let connection/discovery exceptions escape, producing 144 setup errors instead of skips.
- `test_backup_full_help_describes_conditional_default` fails across every supported Python version because Rich inserts the table border `│` between words at line wraps; whitespace normalization does not remove it.
- `test_cli_main_interface` has an unconditional xfail for direct Typer-app invocation. It XPASSes on Python 3.9 and XFAILs on newer versions, while the separate subprocess test already verifies the real no-argument entry point.

Expected behavior: device-dependent tests should skip when no test device exists, and device-independent CLI tests should pass consistently across Python 3.9-3.14.

### How can we reproduce this?

```bash
1. On a Linux host with no iPhone and no `/var/run/usbmuxd`, check out `master` at `fa85df172b5e8876ca14f3a027f1f16704a3e0af`.
2. Install the documented development dependencies: `python -m pip install -U -e '.[test]'`.
3. Run `python -m pytest -ra`.
4. Observe `1 failed, 270 passed, 8 skipped, 1 xfailed, 144 errors`. The errors originate in `tests/conftest.py` when `create_using_usbmux()` raises `ConnectionFailedToUsbmuxdError`.
5. Run `tests/cli/test_backup_cli.py` on Python 3.9 through 3.14; the help assertion fails because a Rich border separates wrapped words.
6. On Python 3.9, `test_cli_main_interface` XPASSes; on newer interpreters it XFAILs.
```

### Environment

Host OS: Ubuntu 24.04.4 LTS, Linux 6.17, ARM64
Python: 3.9.25, 3.10.20, 3.11.15, 3.12.3, 3.13.14, and 3.14.6
pymobiledevice3: master `fa85df172b5e8876ca14f3a027f1f16704a3e0af` (9.36.1.dev3)
Device: none attached (this is the test-fixture condition being reported)
Target OS: N/A

### Relevant log output

```shell
Before fixes on Python 3.12:


FAILED tests/cli/test_backup_cli.py::test_backup_full_help_describes_conditional_default
ERROR ... ConnectionFailedToUsbmuxdError (144 device-dependent setup errors)
== 1 failed, 270 passed, 8 skipped, 1 xfailed, 144 errors ==


The help failure is caused by normalized output containing `full │ │ for an` instead of `full for an`.
```

### Search

- [x] I searched for existing open and closed issues before opening this

### AI assistance

none

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### Additional context

Proposed fixes (already applied and verified on a local audit branch):

1. In `service_provider` and `lockdown`, catch `ConnectionFailedToUsbmuxdError`, `DeviceNotFoundError`, and `NoDeviceConnectedError`, then `pytest.skip("No test device is available through usbmuxd")`. Pairing/protocol failures still remain real errors.
2. Strip Rich's `│` table border before joining wrapped backup help text.
3. Keep `test_cli_main_interface` focused on explicit `--help`; rely on the existing subprocess test for real no-argument behavior.

Verification after the fixes:
- Full Python 3.12 suite: `273 passed, 152 skipped`
- Focused CLI regression set: `7 passed` on each Python 3.9, 3.10, 3.11, 3.12, 3.13, and 3.14
- Ruff check and Ruff format: passed

--- a/tests/cli/test_backup_cli.py
+++ b/tests/cli/test_backup_cli.py
@@ -37,7 +37,9 @@ def test_backup_full_help_describes_conditional_default():
     result = runner.invoke(__main__.app, ["backup2", "backup", "--help"])
 
     assert result.exit_code == 0
-    normalized_output = " ".join(result.output.split())
+    # Rich wraps the help text inside panel borders, so strip the box-drawing
+    # characters before joining lines back into a single string
+    normalized_output = " ".join(result.output.replace("│", " ").split())
     assert "incremental" in normalized_output
     assert "valid local metadata exists" in normalized_output
     assert "full for an" in normalized_output

--- a/tests/cli/test_cli.py
+++ b/tests/cli/test_cli.py
@@ -11,14 +11,13 @@ pytestmark = [pytest.mark.cli]
 ANSI_ESCAPE = re.compile(r"\x1b\[[0-?]*[ -/]*[@-~]")
 
 
-@pytest.mark.xfail(reason="Looks like click broke something")
 def test_cli_main_interface():
+    # No-argument behavior is covered by test_cli_from_python_m_without_args, which
+    # exercises the real entry point; invoking the Typer app directly without
+    # arguments behaves inconsistently across click versions
     runner = CliRunner()
-    r1 = runner.invoke(__main__.app, ["--help"])
-    assert r1.exit_code == 0
-
-    r2 = runner.invoke(__main__.app)
-    assert r2.exit_code == 0
+    result = runner.invoke(__main__.app, ["--help"])
+    assert result.exit_code == 0
 
 
 def test_cli_from_python_m_without_args():

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -5,7 +5,12 @@ from typing import Any, Union
 import pytest
 import pytest_asyncio
 
-from pymobiledevice3.exceptions import DeviceNotFoundError, InvalidServiceError
+from pymobiledevice3.exceptions import (
+    ConnectionFailedToUsbmuxdError,
+    DeviceNotFoundError,
+    InvalidServiceError,
+    NoDeviceConnectedError,
+)
 from pymobiledevice3.lockdown import UsbmuxLockdownClient, create_using_usbmux
 from pymobiledevice3.remote.remote_service_discovery import RemoteServiceDiscoveryService
 from pymobiledevice3.services.dvt.instruments.dvt_provider import DvtProvider
@@ -29,6 +34,16 @@ def pytest_addoption(parser):
         metavar="PATH",
         help="Path to xcuitest JSON config file",
     )
+
+
+NO_DEVICE_SKIP_REASON = "No test device is available through usbmuxd"
+
+
+async def _create_usbmux_client() -> UsbmuxLockdownClient:
+    try:
+        return await create_using_usbmux()
+    except (ConnectionFailedToUsbmuxdError, DeviceNotFoundError, NoDeviceConnectedError):
+        pytest.skip(NO_DEVICE_SKIP_REASON)
 
 
 @pytest.fixture(scope="function")
@@ -75,7 +90,7 @@ async def service_provider(
             for rsd in rsds:
                 await rsd.close()
     else:
-        async with await create_using_usbmux() as client:
+        async with await _create_usbmux_client() as client:
             yield client
 
 
@@ -96,7 +111,7 @@ async def lockdown() -> AsyncGenerator[UsbmuxLockdownClient, Any]:
     """
     Creates a new lockdown client for each test.
     """
-    async with await create_using_usbmux() as client:
+    async with await _create_usbmux_client() as client:
         yield client
 
 

